token/core/tms.go

```diff
@@ -47,10 +47,14 @@ type TMSProvider struct {
 	services map[string]driver.TokenManagerService
 }
 
-func NewTMSProvider(logger logging.Logger, configProvider ConfigProvider, pps PublicParametersStorage, tokenDriverService *TokenDriverService) *TMSProvider {
+func NewTMSProvider(
+	configService ConfigService,
+	pps PublicParametersStorage,
+	tokenDriverService *TokenDriverService,
+) *TMSProvider {
 	ms := &TMSProvider{
-		logger:                  logger,
-		configProvider:          configProvider,
+		logger:                  logging.MustGetLogger(),
```

Comment:
should this logger get a specific name? this code now attaches the package logger to the TMS, what is the actual idea to give to the TMS provider a special instance rather than accessing the logger through the package logger?

Maybe this is is a separate discussion - we used to use package based loggers - only very special situation, we were creating a specific logger and attached to a struct.

Reply:
agreed